### ISSUE TITLE
Uncaught exception when moving FromScratch Window on the desktop

### ISSUE THREAD

**Issue by notbalanced** [Kilian/fromscratch#41]
Hi, 

I get the following pop up error when I move the FromScratch window around on the desktop (Windows 7/10 x64).  It doesn't crash the application, and works fine otherwise.

```
[Window Title]
Error

[Main Instruction]
A JavaScript error occurred in the main process

[Content]
Uncaught Exception:
Error: EPERM: operation not permitted, rename 'C:\Users\xxxx\.fromscratch\windowstate.2676887468' -> 'C:\Users\xxxx\.fromscratch\windowstate'
    at Error (native)
    at Object.fs.renameSync (fs.js:809:18)
    at writeFileSync (C:\Util\FromScratch\resources\app.asar\node_modules\write-file-atomic\index.js:81:8)
    at JSONStorage.LocalStorage.setItem (C:\Util\FromScratch\resources\app.asar\node_modules\node-localstorage\LocalStorage.js:177:7)
    at JSONStorage.setItem (C:\Util\FromScratch\resources\app.asar\node_modules\node-localstorage\LocalStorage.js:289:44)
    at storeWindowState (C:\Util\FromScratch\resources\app.asar\main.js:129:24)
    at BrowserWindow.mainWindow.on (C:\Util\FromScratch\resources\app.asar\main.js:134:7)
    at emitOne (events.js:96:13)
    at BrowserWindow.emit (events.js:188:7)

[OK]
```

**Comment by Kilian** [comment]
Can you try: 

1. Closing FromScratch,
2. Delete the file 'C:\Users\xxxx\.fromscratch\windowstate'
3. Open FromScratch

?

**Comment by notbalanced** [comment]
Hi Killian, 

Thanks for the reply. I deleted the windowstate file and reopened FromScratch. However, it still occurs when I try to drag the FromScratch window to a new location on the screen.

It is not a big deal, and doesn't prevent me from using the app (which is great, BTW). But thought you'd want to be aware of the issue. 

(I'm using 1.2.0.)

**Comment by Kilian** [comment]
Then it's probably a rights issue. Can you check if that .fromscratch folder is owned by your user? I'm not sure how that works in Windows.

**Comment by Kilian** [comment]
@notbalanced Has this been resolved by setting the correct rights?

**Comment by notbalanced** [comment]
Hi Killian, 

I checked the directory rights and those are set correctly. The issue still occur when I do something to change the window state (like minimize or move the window). It seems like maybe a timing where it is trying to write the file too soon. But I'm just guessing... It isn't a showstopper.

**Comment by Kilian** [comment]
Then maybe the windowstate file is used/open/reserved in some other application?

**Comment by notbalanced** [comment]
Hi,
I thought it might help to show an animated gif of the error. I'm not accessing any of the files in another app: text editor or anything like that. Possibly Antivirus? But I don't know. It only happens as I move the window around.
![fromscratcherror](https://cloud.githubusercontent.com/assets/698560/21958610/14f24c1e-da80-11e6-8e04-ed4c49739e9a.gif)

**Comment by mistapetroz** [comment]
First of all, Awesome up! ... This error also occurs when resizing the window also.

**Comment by Kilian** [comment]
Any resize, move or close action triggers the same action, so at least it's consistent. I'll see what I can figure out on a windows instance!

**Comment by Kilian** [comment]
This appears to have been fixed recenty in the library we use to store data: https://github.com/lmaccherone/node-localstorage/issues/23 so will be fixed in the next release

**Comment by spike1234** [comment]
I dont' want to push, but can I ask when will be the next release?

**Comment by Kilian** [comment]
@spike1234 I can't say; This is a sideproject for me and there are a couple of things I want to add before an official next release (primarily: installers). All the changes are already in master, so you can compile your own (see readme), but that does require some setup and is not the most userfriendly.

**Comment by spike1234** [comment]
ok, thanks

**Comment by Kilian** [comment]
Should be released in the next couple of days as part of 1.4.0 :)
